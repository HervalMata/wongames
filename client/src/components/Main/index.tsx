import * as S from './styles'

const Main = ({title = 'Won Games', description = 'TypeScript, ReactJS, NextJS e StyledComponents'}) => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="Imagem de um átomo e react avançado ao lado." />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration src="/img/hero-illustration.svg" alt="Um desenvolvedor de frente para uma tela com código." />
  </S.Wrapper>
)

export default Main
