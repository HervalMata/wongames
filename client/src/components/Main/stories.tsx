import Main from "./index";
import {text, withKnobs} from "@storybook/addon-knobs";
import {Meta, Story} from "@storybook/react/types-6-0";

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
} as Meta

export const Basic : Story = () => (
  <Main
    title={text('Title', 'Won Games')}
    description={text(
    'description',
      'TypeScript, ReactJS, NextJS e StyledComponents'
    )}
  />
)
