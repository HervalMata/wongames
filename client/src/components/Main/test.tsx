import { render, screen } from "@testing-library/react";

import Main from "./index";

describe('<Main />', () => {
  it('should render the heading', function () {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /won games/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  });
})
