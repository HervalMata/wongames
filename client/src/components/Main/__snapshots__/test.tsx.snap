// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-1cs6c8-0 dbMGZu"
>
  <img
    alt="Imagem de um átomo e react avançado ao lado."
    class="styles__Logo-sc-1cs6c8-1 hWruYj"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1cs6c8-2 cEWuhb"
  >
    Won Games
  </h1>
  <h2
    class="styles__Description-sc-1cs6c8-3 jCSlkc"
  >
    TypeScript, ReactJS, NextJS e StyledComponents
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="styles__Illustration-sc-1cs6c8-4 ePmPKR"
    src="/img/hero-illustration.svg"
  />
</main>
`;
